### ISSUE TITLE
BatchNormalization axis=0 fails on batch dim changes.

### ISSUE THREAD

**Issue by cottrell** [tensorflow/tensorflow#31027]
Will close if this ends up to not be a bug. Posted on stackoverflow now and will follow up later.

https://stackoverflow.com/questions/57202668/keras-batchnormalization-only-works-for-constant-batch-dim-when-axis-0

**Comment by cottrell** [comment]
Probably axis meaning is the confusing bit and intention is not what axis to average over but what axis *not* to average over. Not sure if broadcast across other axes.
